src/interop/js-interop/usage.md

```diff
@@ -0,0 +1,368 @@
+```
+
+Calling `array.push(0.toJS)` in Dart will result in a JS invocation of
+`array.push(0.toJS)` and *not* `array.push(0.toJS, null)`. If you declare a
+parameter with an explicit default value, you will get a warning that the value
+will be ignored.
+
+## `@JS()`
+
+It is sometimes useful to refer to a JS property with a different name than the
+one written. For example, if you want to write two `external` APIs that point to
+the same JS property, you’d need to write a different name for at least one of
+them. In order to do this, you can use the [`@JS()`] annotation with a constant
+string value. For example:
+
+```dart
+extension type Array._(JSArray<JSAny?> _) implements JSArray<JSAny?> {
+  external int push(JSNumber number);
+  @JS('push')
+  external int pushString(JSString string);
+}
+```
+
+Calling either `push` or `pushString` will result in JS code that uses `push`.
+
+You can also rename interop types:
+
+```dart
+@JS('Date')
+extension type JSDate._(JSObject _) implements JSObject {
+  external JSDate();
+
+  external static int now();
+}
+```
+
+Calling `JSDate()` will result in a JS invocation of `new Date()`. Similarly,
+calling `JSDate.now()` will result in a JS invocation of `Date.now()`.
+
+Furthermore, you can namespace an entire library, which will add a prefix to all
+interop top-level members, interop types, and `static` interop members within
+those types:
+
+```dart
+@JS('library1')
+library;
+
+import 'dart:js_interop';
+
+@JS()
+external void method();
+
+extension type JSType(JSObject _) implements JSObject {
+  external JSType();
+
+  external static int get staticMember;
+}
+```
+
+Calling `method()` will result in a JS invocation of `library1.method()`,
+calling `JSType()` will result in a JS invocation of `new library1.JSType()`,
+and calling `JSType.staticMember` will result in a JS invocation of
+`library1.JSType.staticMember`.
+
+Unlike interop members and interop types, we only ever add a library name in the
+JS invocation if you provide a non-empty value in the `@JS()` annotation on the
+library.
+
+```dart
+library interop_library;
+
+import 'dart:js_interop';
+
+@JS()
+external void method();
+```
+
+Calling `method()` will result in a JS invocation of `method()` and *not*
+`interop_library.method()`.
+
+You can also write multiple namespaces delimited by a '.' for libraries,
+top-level members, and interop types:
+
+```dart
+@JS('library1.library2')
+library;
+
+import 'dart:js_interop';
+
+@JS('library3.method')
+external void method();
+
+@JS('library3.JSType')
+extension type JSType(JSObject _) implements JSObject {
+  external JSType();
+}
+```
+
+Calling `method()` will result in a JS invocation of
+`library1.library2.library3.method()`, calling `JSType()` will result in a JS
+invocation of `new library1.library2.library3.JSType()`, and so forth.
+
+You can't use `@JS()` annotations with '.' in the value on interop type members
+or extension members of interop types, however.
+
```

Comment:
Maybe you should either add a sub heading here or at least an anchor. Not that you _need_ to reference it earlier in the annotation section (even though I was wondering), but because the empty `@JS()` comes up way at the top of the page and maybe would be useful to link to:

```md
### Empty `@JS()`
```

Or
```md
<a id="empty-js"></a>
```

And then somewhere up in the [Top-level members section](https://github.com/dart-lang/site-www/pull/4578/files#diff-173da65856efe1cfdb286777d80c04a313dad6171f415b1403693da8f8a95e29R52), add a sentence saying "Top level members must be prepended with an [empty `@JS()` annotation]()" and that links to (#empty-js)